### PR TITLE
disallow setting profile names with underscore

### LINKED ISSUE #8493: profile name should not have "_"
```
Invalid value: "vm_0_10_centos": a DNS-1123 subdomain must consist of lower case alphanumeric character
```

### PR DESCRIPTION
**Pull request by jlucktay**
Fixes #8493.

### CODE REVIEW

**Comment by jlucktay** [comment]
The linked [pull request process documentation](https://git.k8s.io/community/contributors/guide/pull-requests.md) has a few bullet point TODOs in the [Run Local Verifications](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#run-local-verifications) section:

> - Run and pass `make verify` (can take 30-40 minutes)
> - Run and pass `make test`
> - Run and pass `make test-integration`

However, the Makefile doesn't have targets for `verify` nor `test-integration`.

File: pkg/minikube/config/profile_test.go [hunk 1]
```diff
@@ -73,24 +73,37 @@ func TestListProfiles(t *testing.T) {
...
+		"1st-2nd-3rd-profile": true,
+		"-profile":            false,
+		"meaningful_name":     false,
+		"meaningful_name@":    false,
+		"n_a_m_e_2":           false,
+		"n":                   true,
+		"_name":               false,
+		"N__a.M--E12567":      false,
+		"1":                   true,
+		"12567":               true,
```
Comment (medyagh) [inline pkg/minikube/config/profile_test.go:93]: Thank you for doing this PR for better readoablity I suggest moving all the false ones under each other

Comment (jlucktay) [reply pkg/minikube/config/profile_test.go:85]: 85a641177 resolves

**Pull request merged**